payu/models/cesm.py

```diff
@@ -0,0 +1,248 @@
+# coding: utf-8
+"""
+The payu interface for CESM model configurations using CMEPS/NUOPC
+-------------------------------------------------------------------------------
+Contact: Marshall Ward <marshall.ward@anu.edu.au>
+-------------------------------------------------------------------------------
+Distributed as part of Payu, Copyright 2011 Marshall Ward
+Licensed under the Apache License, Version 2.0
+http://www.apache.org/licenses/LICENSE-2.0
+"""
+from __future__ import print_function
+
+# Standard Library
+import os
+import re
+import glob
+import shutil
+import multiprocessing
+
+# Local
+from payu.fsops import mkdir_p
+from payu.models.model import Model
+
+component_info = {
+    "mom6": {
+        "realm": "ocn",
+        "config_files": [
+            "input.nml",
+            "MOM_input",
+            "diag_table",
+        ],
+        "optional_config_files" : ["MOM_override"]
+    },
+    "cice6": {
+        "realm": "ice",
+        "config_files": ["ice_in"],
+    },
+    "ww3": {
+        "realm": "wav",
+        "config_files": ["wav_in"],
+    },
+    "datm": {
+        "realm": "atm",
+        "config_files": [
+            "datm_in",
+            "datm.streams.xml",
+        ],
+    },
+    "drof": {
+        "realm": "rof",
+        "config_files": [
+            "drof_in",
+            "drof.streams.xml"
+        ]
+    },
```

Comment:
@dougiesquire payu dies with `KeyError: 'docn'` when running https://github.com/COSIMA/CICE6-WW3 so I guess we need something like this here?
```
    "docn": {
        "realm": "ocn",
        "config_files": [
            "docn_in",
            "docn.streams.xml",
        ],
    },
```

Reply:
Yes, sorry, I had the change locally but hadn't pushed it. Pip installing from this PR again should include the required change.

Reply:
great, thanks!